language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1624,6 +1624,29 @@ public static boolean isSelfClassSymbol(Symbol symbol, PositionedOperationContex
         return classSymbol.equals(varTypeSymbol);
     }
 
+    /**
+     * Check if the symbol is an object symbol with self as the name.
+     *
+     * @param symbol               Symbol
+     * @param nodeAtCursor         Node
+     * @return {@link Boolean} whether the symbol is a self object symbol.
+     */
+
+    public static boolean isSelfObjectSymbol(Symbol symbol, Node nodeAtCursor) {
+        Node currentNode = nodeAtCursor;
```

Comment:
Can we use  the `nodeAtCursor` variable instead of declaring a new variable `currentNode`? WDYT? @mindula  @IMS94